### DISCUSSION TITLE
How can I disable "Details Library Biography" Rating buttons?

### DISCUSSION THREAD

**Discussion by pityalvarez** [TT-ReBORN/Georgia-ReBORN#48] [Q&A] [answered]
Except Lyrics button, but if not, disable the five buttons and only that.
And i noticed that play or pause on the album cover its not instant, i don't know if it's coded like this, I'd like it snappy

**Comment by TT-ReBORN** [comment]
Hi,

You can't remove the top menu buttons without modifying javascript files, why would you do it anyways?
I'm in the middle of rewriting and refactoring lots of code in the theme so I won't give you how to do instructions now,
because the top menu code was also refactored. Come again in a week or two, once I have released the update.

-TT

**Comment by pityalvarez** [comment]
I want to do it because I think that visually it would look cleaner (whenever we set things up our way it always looks that way, the same is for everyone), it would also reduce the loading time at startup because it does not have 4 sections of utilities that I do not use. I thought it was done on purpose that there is an option to change fonts and sizes but not to change the size of the bottom bar or album cover, it was ignorant of me.
My requests/comments are complicated to apply, and I have no knowledge of the work of a programmer. 
I would also like to know how to get updates notifications in any possible way, I can't find a way how to get them.
Thanks for the response.

**Reply by TT-ReBORN** [reply]
Hi @pityalvarez,

I have not forgot about your request but I'm still working on the update. I'll promise that once the update has been released I give you detailed instructions how to do it, because right now, the old code would be too inconvenient to modify for you...

Once the update has been released, I will notify you here with instructions. All I can say right now is that the wait will be worth it!
Thanks for the understanding.

-TT

**Comment by TT-ReBORN** [comment, marked as answer]
@pityalvarez

sorry for the long wait, I've just released the update:
https://hydrogenaud.io/index.php?topic=121047.msg1010153#msg1010153

Download and install it on a new fresh portable installation, open foobar2000\profile\georgia-reborn\js\georgia-reborn-main.js  
and go to line 5430:

just add `x = 9999;`

it should look like this:

```
x = 9999;

x += img[0].Width - (is_4k ? 3 : 2);
img = btnImg.Playlist;
btns.playlist = new Button(x, y, img[0].Width, h, 'Playlist', img, 'Display Details');

x += img[0].Width - (is_4k ? 3 : 2);
img = btnImg.ShowLibrary;
btns.library = new Button(x, y, img[0].Width, h, 'ShowLibrary', img, 'Display Library');

x += img[0].Width - (is_4k ? 3 : 2);
img = btnImg.Biography;
btns.biography = new Button(x, y, img[0].Width, h, 'Biography', img, 'Display Biography');

x += img[0].Width - (is_4k ? 3 : 2);
img = btnImg.Lyrics;
btns.lyrics = new Button(x, y, img[0].Width, h, 'Lyrics', img, 'Display Lyrics');

x += img[0].Width - (is_4k ? 3 : 2);
img = btnImg.Rating;
btns.rating = new Button(x, y, img[0].Width, h, 'Rating', img, 'Rate Song');
```

save it and restart foobar.

If you want to display lyrics without the buttons, just right click on album art > Display lyrics. Right click again > Hide lyrics.

-TT

**Reply by pityalvarez** [reply]
Loved the update, like you said "You can literally spend hours in experimenting and achieving", it took me time to be satisfied to my taste with so many options, it made me dizzy haha. I love the Black theme with blend 2, it's the effect I was waiting for because it looks like the qobuz store. But I'm trying to figure out a bug I have which is that using the space bar to pause shows the same box that is shown when you type letters, from an option that I don't know the truth, in these days maybe I'll fix it on my own.
I explained myself wrong before, because while experimenting I saw that there are options to disable things (I don't remember well what, but if you disable "default" it would stop showing the respective), and I was hoping that in the future there will be an option to disable the options above so the interface looks more minimalist. I don't think coding it is easy, just in my imagination it sounded great to apply. I tried the option you made for me, I appreciate it, but with what I planned before, I just thought some option like that was possible, for everyone.
![image](https://user-images.githubusercontent.com/97750378/167347671-0443cdda-7aa4-4948-8532-1abe8ea3ab1b.png)

**Reply by TT-ReBORN** [reply]
@pityalvarez,

if you hit some keys on your keyboard, the jump search will be activated. In your case, you hit the space bar and it will show an empty black rectangle instead of a letter. Well if you want to deactivate it, you can open and modify the javascript file ui-components.js and go to this line here ( line 647 or copy and search ):

`if (utils.IsKeyPressed(0x09) || utils.IsKeyPressed(0x11) || utils.IsKeyPressed(0x1B) || utils.IsKeyPressed(0x6A) || utils.IsKeyPressed(0x6D)) return;`

and replace it with this:

`if (utils.IsKeyPressed(0x09) || utils.IsKeyPressed(0x11) || utils.IsKeyPressed(0x1B) || utils.IsKeyPressed(0x6A) || utils.IsKeyPressed(0x6D) || utils.IsKeyPressed(0x20)) return; `

After you saved the file and reloaded the theme, it will ignore space bar strokes.

-TT
